src/tests/miscellaneous/mock-adb/app/port-forward-server.js

```diff
@@ -4,30 +4,82 @@ const child_process = require('child_process');
 const fs = require('fs');
 const path = require('path');
 const process = require('process');
+const { startMockService } = require('../../mock-service-for-android/mock-service');
 
 // This is helpful for debugging, but causes us to not be able to exit before "detached" child
 // process (the "correct" behavior), so we leave it off except when debugging mock-adb itself
 const LOG_DETACHED_PROCESS = false;
 
 const portForwardSentinelArgument = '__PORT_FORWARD_SERVER';
+const portForwardProcessReadyMessage = 'PORT_FORWARD_PROCESS_READY';
+const portForwardProcessStartTimeoutMs = 10000;
 
 function portFile(port, fileExtension) {
     return path.join(path.dirname(process.execPath), `test-server-${port}.${fileExtension}`);
 }
 
-function startDetachedPortForwardServer(port, path) {
-    const testServerProcess = child_process.spawn(
-        process.argv[0],
-        [process.argv[1], portForwardSentinelArgument, port, path],
+// Returns whether argv indicated it *should* be handled as a port server
+async function tryHandleAsPortForwardServer(argv) {
+    if (argv[2] !== portForwardSentinelArgument) {
+        return false;
+    }
+
+    const port = parseInt(process.argv[3], 10);
+    const path = process.argv[4];
+
+    await startMockService(port, path);
+
+    process.send(portForwardProcessReadyMessage);
```

Comment:
I verified that the message was being waited on by making temporary modifications to this line to send the wrong message and verifying that mock-adb started exhibiting the new timeout behavior.